lib/Parser/Parse.cpp

```diff
@@ -915,10 +916,21 @@ Symbol* Parser::AddDeclForPid(ParseNodePtr pnode, IdentPtr pid, SymbolType symbo
                 Error(ERRRedeclaration);
             }
         }
+        else if (!pnodeFnc->sxFnc.IsBodyAndParamScopeMerged()
+            && scope->GetScopeType() == ScopeType_FunctionBody
+            && (pnode->nop == knopLetDecl || pnode->nop == knopConstDecl))
+        {
+            // In case of split scope function when we add a new let or const declaration to the body
+            // we have to check whether the param scope already has the same symbol defined.
+            PidRefStack *paramScopeRef = pid->GetPidRefForScopeId(pnodeFnc->sxFnc.pnodeScopes->sxBlock.blockId);
+            if (paramScopeRef && paramScopeRef->GetSym() && !paramScopeRef->GetSym()->GetIsArguments())
```

Comment:
on the ref stack if the node is just a reference to the pid, not a declaration, then sym will be null but the refstack will have the scopeid and function id. We later bind the right symbol to the ref stack. If it is a declaration then sym won't be null.